### ISSUE TITLE
Rename "language" to "generator-name" in cli and maven-plugin

### ISSUE THREAD

**Issue by jmini** [OpenAPITools/openapi-generator#39]
On the long term we might want to offer an other way to select the targeted generator.

Probably with multiple switches (to be defined more precisely):
* language
* generator type 
* framework
* ...

Examples:
```
cli -type=client -language=kotlin
cli -type=server -language=kotlin -framework=ktor
```

A first move that we could do now with our first release is to rename the term “language” with “generator-name“ in the maven Plugin and in the CLI.

We might support them both options for backward compatibility, but with a warning explaining that “generator-name“ should be preferred.

What are the opinions?

cc: @jimschubert

**Comment by jimschubert** [comment]
I agree with supporting both but showing a deprecation message on language. I would suggest erroring if both are set to avoid potential issues. I'd also suggest clearly defining the removal timeline (maybe by version 3.1).

Additionally, while we deprecate the language option, we could also provide a mapping of the generator renames. This would allow users to almost seamlessly upgrade, and the CLI would notify of the necessary changes for subsequent releases.

**Comment by jimschubert** [comment]
And to the split of options, I fully agree with this. I haven't yet made such a change, because I'd consider this too much of a breaking change to include in the 3.0.0 release. We'll probably want to target this for 4.0.0.

**Comment by jmini** [comment]
Thank you for this feedback, I like all theses suggestions.

One might be more complicated:
> I'd also suggest clearly defining the removal timeline (maybe by version 3.1).

I am not sure if we have a release roadmap/timeline by now...

**Comment by jimschubert** [comment]
Even without a roadmap, I think we can explicitly say "we will remove this in the next minor release". I think my use of "timeline" to refer to the minor release was confusing. My point was that we shouldn't randomly remove it in a build revision (e.g. 3.0.22), as things like this catch people off guard.

**Comment by jimschubert** [comment]
Oh, I see what you were saying about removing language. If we were to move to `type`, `language` and `framework` then we couldn't actually remove language?

It's likely that we'd want to go through a full major release with `generator-name` before reusing `language` for the actual language name.

**Comment by jimschubert** [comment]
@jmini I've opened #57 with the intro of generator-name and deprecation of 'language' as a CLI and Maven Plugin option.  I think this is a code quality thing that should make it into the 3.0.0 release so that the deprecation can exist for an entire major release.

I've just thought of a question about the implementation. I'll add it there.

**Comment by jmini** [comment]
Thank you a lot for opening the PR, and yes it should be part of our first 3.0.0 release.

**Comment by jimschubert** [comment]
@jmini #57 was merged. Do you still want to keep this issue open for the discussion about breaking things down to language/type/framework?

**Comment by jmini** [comment]
No I think we should close this one as solved with version `3.0.0`, and continue with #137.
